paymentsheet/src/main/java/com/stripe/android/paymentsheet/forms/FormViewModel.kt

```diff
@@ -72,12 +82,94 @@ internal class FormViewModel @Inject internal constructor(
             merchantName = formArguments.merchantName,
             context = context,
             shippingValues = formArguments.shippingDetails
-                ?.toIdentifierMap(formArguments.billingDetails)
+                ?.toIdentifierMap(formArguments.billingDetails),
         ).transform(
-            requireNotNull(lpmRepository.fromCode(formArguments.paymentMethodCode)).formSpec.items
+            modifiedSpecsWithBillingDetails(
```

Comment:
I'm thinking it might be worth it to extract the billing details specific stuff into it's own class, and test it all in isolation.

Reply:
Good idea, moved it